homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -133,15 +139,39 @@ class EnvoyProductionSensorEntityDescription(
         suggested_unit_of_measurement=UnitOfEnergy.MEGA_WATT_HOUR,
         suggested_display_precision=3,
         value_fn=lambda production: production.watt_hours_lifetime,
+        on_phase=None,
     ),
 )
 
 
+def phase_sensor_from_production_sensor(
+    sensors: list[EnvoyProductionSensorEntityDescription],
+) -> dict[str, list[EnvoyProductionSensorEntityDescription]]:
+    """Build phase sensors from production sensors."""
+    return {
+        (PhaseNames(PHASENAMES[phase])): [
+            replace(
+                sensor,
+                key=f"{sensor.key}_l{phase + 1}",
+                translation_key=f"{sensor.translation_key}_phase",
+                on_phase=PhaseNames(PHASENAMES[phase]),
+                translation_placeholders={"phase_name": f"l{phase + 1}"},
+            )
+            for sensor in sensors
+        ]
+        for phase in range(0, 3)
+    }
+
+
+PRODUCTION_PHASE_SENSORS = phase_sensor_from_production_sensor(list(PRODUCTION_SENSORS))
```

Comment:
I think we can drop the function here and build the constant directly

Reply:
This works:
```py
PRODUCTION_PHASE_SENSORS = {
    (PhaseNames(PHASENAMES[phase])): [
        replace(
            sensor,
            key=f"{sensor.key}_l{phase + 1}",
            translation_key=f"{sensor.translation_key}_phase",
            on_phase=PhaseNames(PHASENAMES[phase]),
            translation_placeholders={"phase_name": f"l{phase + 1}"},
        )
        for sensor in list(PRODUCTION_SENSORS)
    ]
    for phase in range(0, 3)
}
```

Reply:
I think you could write it as
```python
PRODUCTION_PHASE_SENSORS = {
    (on_phase := PhaseNames(PHASENAMES[phase])): [
        replace(
            sensor,
            key=f"{sensor.key}_l{phase + 1}",
            translation_key=f"{sensor.translation_key}_phase",
            on_phase=on_phase,
            translation_placeholders={"phase_name": f"l{phase + 1}"},
        )
        for sensor in PRODUCTION_SENSORS
    ]
    for phase in range(0, 3)
}
```

Reply:
yep, works fine as well.